web-local/src/lib/components/navigation/dashboards/MetricsDefinitionAssets.svelte

```diff
@@ -57,6 +58,18 @@
   let showRenameMetricsDefinitionModal = false;
   let renameMetricsDefName = null;
 
+  async function getModelFromDashboardName(
+    instanceId: string,
+    metricViewName: string
+  ) {
+    const metricViewCatalog = await runtimeServiceGetCatalogEntry(
```

Comment:
This would disable this when metrics is invalid. I think we should instead use the yaml for this. We can perhaps cache the yaml's json representation in a store. `file-artifact-store` might be a good place. We could reuse it for other entities.